src/main/java/io/druid/data/input/impl/JSONParseSpec.java

```diff
@@ -49,18 +59,47 @@ public void verify(List<String> usedCols)
   @Override
   public Parser<String, Object> makeParser()
   {
-    return new JSONParser(objectMapper, null);
+    if(flattenSpec.isEnabled()) {
```

Comment:
can this just be flattenSpec != null ? and let the passed flattenSpec come through?